impl/src/main/java/slack/ReactionMessageHandler.java

```diff
@@ -42,26 +44,30 @@
     private final        AnswerResource     answerResource;
     private final        UserAnswerResource userAnswerResource;
     private final        SlackResource      slackResource;
-    private final        UserResource       userResource;
 
     @Inject
     public ReactionMessageHandler(SlackResource slackResource,
-        UserResource userResource,
         QuestionResource questionResource,
         AnswerResource answerResource,
         UserAnswerResource userAnswerResource
     ) {
 
         this.slackResource = slackResource;
-        this.userResource = userResource;
         this.questionResource = questionResource;
         this.answerResource = answerResource;
         this.userAnswerResource = userAnswerResource;
     }
 
     @Override
     public boolean shouldHandle(String type, JsonObject body) {
-        return HANDLED_TYPES.contains(type) && MESSAGE.equals(body.get(ITEM).getAsJsonObject().get(TYPE).getAsString());
+        String itemType = Optional.ofNullable(body)
+            .map(jsonObject -> jsonObject.get(ITEM))
```

Comment:
When using constants instead of strings, I expect them to explain the meaning of the string rather than being the same but upper case.
What is the point of constants like these?

Reply:
The main point, as I see it, is reusability in implementation and tests. Naming could probably be improved, but clarity was not my main purpose here, although I do think it looks "better" with constants.